mycli/main.py

```diff
@@ -690,15 +690,39 @@ def output(self, output, status=None):
 
         """
         if output:
-            self.log_output(output)
-            special.write_tee(output)
-            special.write_once(output)
-
-            if (self.explicit_pager or
-                    (special.is_pager_enabled() and not self.output_fits_on_screen(output, status))):
-                click.echo_via_pager(output)
-            else:
-                click.secho(output)
+            size = self.cli.output.get_size()
+
+            margin = self.get_output_margin(status)
+
+            fits = True
+            buf = []
+            output_via_pager = self.explicit_pager and special.is_pager_enabled()
+            for i, line in enumerate(output, 1):
+                self.log_output(line)
+                special.write_tee(line)
+                special.write_once(line)
+
+                if fits or output_via_pager:
```

Comment:
If the output doesn't fit and it's not going to be output via a pager, then we lose all the rows after `fits` is set to `False`.

This is the case when the pager is disabled.